connector/src/main/java/org/jboss/as/connector/services/workmanager/statistics/WorkManagerStatisticsService.java

```diff
@@ -0,0 +1,134 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.connector.services.workmanager.statistics;
+
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Map;
+
+import org.jboss.as.connector.dynamicresource.ClearWorkManagerStatisticsHandler;
+import org.jboss.as.connector.subsystems.resourceadapters.CommonAttributes;
+import org.jboss.as.connector.subsystems.resourceadapters.Constants;
+import org.jboss.as.connector.subsystems.resourceadapters.WorkManagerRuntimeAttributeReadHandler;
+import org.jboss.as.connector.subsystems.resourceadapters.WorkManagerRuntimeAttributeWriteHandler;
+import org.jboss.as.controller.OperationStepHandler;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.ResourceBuilder;
+import org.jboss.as.controller.SimpleAttributeDefinition;
+import org.jboss.as.controller.descriptions.OverrideDescriptionProvider;
+import org.jboss.as.controller.descriptions.StandardResourceDescriptionResolver;
+import org.jboss.as.controller.registry.ManagementResourceRegistration;
+import org.jboss.dmr.ModelNode;
+import org.jboss.jca.core.api.workmanager.WorkManager;
+import org.jboss.msc.inject.Injector;
+import org.jboss.msc.service.Service;
+import org.jboss.msc.service.StartContext;
+import org.jboss.msc.service.StartException;
+import org.jboss.msc.service.StopContext;
+import org.jboss.msc.value.InjectedValue;
+
+public class WorkManagerStatisticsService implements Service<ManagementResourceRegistration> {
+
+    private final ManagementResourceRegistration overrideRegistration;
+    private final boolean statsEnabled;
+
+    protected final InjectedValue<WorkManager> workManager = new InjectedValue<>();
+
+
+    /**
+     * create an instance *
+     */
+    public WorkManagerStatisticsService(final ManagementResourceRegistration registration,
+                                        final String name,
+                                        final boolean statsEnabled) {
+        super();
+        overrideRegistration = registration.registerOverrideModel(name, new OverrideDescriptionProvider() {
+            @Override
+            public Map<String, ModelNode> getAttributeOverrideDescriptions(Locale locale) {
+                return Collections.emptyMap();
+            }
+
+            @Override
+            public Map<String, ModelNode> getChildTypeOverrideDescriptions(Locale locale) {
+                return Collections.emptyMap();
+            }
+
+        });
+        this.statsEnabled = statsEnabled;
```

Comment:
looking at this code, statistics can only be set on reload/restart.
could we make it work so that :write-attribute(name=statistics-enabled, value=true) would runtime update this service and enable statistics?
You would need write attribute handler that would call new method this service to enable stats.